tests/test_github.py

```diff
@@ -248,9 +248,11 @@ def test_report_pull_status(self):
         self.waitUntilSettled()
         # We should have a status container for the head sha
         self.assertIn(A.head_sha, A.statuses.keys())
-        # We should only have one status for the head sha
-        self.assertEqual(1, len(A.statuses[A.head_sha]))
-        check_status = A.statuses[A.head_sha][0]
+        # We should only have one status for the head sha but adding a
+        # second pipeline with the same trigger but differen project
+        # currently results in both pipelines reporting status
```

Comment:
Someone in the #zuul IRC channel on Freenode pointed me at a patch I that was on in v2.5.2 of zuul missing from this branch that resolves this part of the patch.